src/commands/scaling/addScaleRule/GetScaleRuleNameStep.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerApp, ContainerAppsAPIClient, ScaleRule } from '@azure/arm-appcontainers';
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { createContainerAppsAPIClient } from '../../../utils/azureClients';
+import { getResourceGroupFromId } from '../../../utils/azureUtils';
+import { localize } from '../../../utils/localize';
+import { IAddScaleRuleWizardContext } from './IAddScaleRuleWizardContext';
+
+export class GetScaleRuleNameStep extends AzureWizardPromptStep<IAddScaleRuleWizardContext> {
+    public hideStepCount: boolean = true;
+    containerApp: ContainerApp | undefined;
```

Comment:
You get the containerApp in multiple places. Seems like you could just save it on the context.